mis_builder/models/mis_report_instance.py

```diff
@@ -527,43 +554,39 @@ def _onchange_dates(self):
                     self.date_to != self.date_range_id.date_end:
                 self.date_range_id = False
 
-    @api.multi
-    def preview(self):
-        self.ensure_one()
-        view_id = self.env.ref('mis_builder.'
-                               'mis_report_instance_result_view_form')
-        return {
-            'type': 'ir.actions.act_window',
-            'res_model': 'mis.report.instance',
-            'res_id': self.id,
-            'view_mode': 'form',
-            'view_type': 'form',
-            'view_id': view_id.id,
-            'target': 'current',
-        }
-
     @api.multi
     def print_pdf(self):
         self.ensure_one()
+        context = dict(self.env.context)
+        context1 = {'active_ids': self.ids}
+        if isinstance(context1, pycompat.string_types):
+            context1 = safe_eval(context1, dict(context))
+        context.update(context1)
         return {
             'name': 'MIS report instance QWEB PDF report',
             'model': 'mis.report.instance',
-            'type': 'ir.actions.report.xml',
+            'type': 'ir.actions.report',
             'report_name': 'mis_builder.report_mis_report_instance',
             'report_type': 'qweb-pdf',
-            'context': self.env.context,
+            'context': context,
         }
 
     @api.multi
     def export_xls(self):
         self.ensure_one()
+        context = dict(self.env.context)
+        context1 = {'active_ids': self.ids}
+        if isinstance(context1, pycompat.string_types):
+            context1 = safe_eval(context1, dict(context))
+        context.update(context1)
```

Comment:
I don't understand this piece. Perhaps `context = dict(self.context, active_ids=self.ids)` ?